impeller/renderer/backend/metal/allocator_mtl.h

```diff
@@ -7,17 +7,38 @@
 
 #include <Metal/Metal.h>
 
+#include "impeller/base/thread.h"
 #include "impeller/core/allocator.h"
 
 namespace impeller {
 
+class DebugAllocatorStats {
+ public:
+  DebugAllocatorStats() {}
+
+  ~DebugAllocatorStats() {}
+
+  void Increment(size_t size);
+
+  void Decrement(size_t size);
+
+  size_t GetAllocationSizeMB();
+
+ private:
+  mutable Mutex mutex_ = {};
+  size_t size_ IPLR_GUARDED_BY(mutex_) = 0;
```

Comment:
There no need for the mutex here, you can use `std::atomic`.  You'd use `fetch_add` and `fetch_sub`.  I'd then assert that the value never goes below zero.

Reply:
good idea

Reply:
I'm not really sure why I'd need fetch_add/fetch_sub over += or -=?

Reply:
> I'm not really sure why I'd need fetch_add/fetch_sub over += or -=?

The unwritten convention inside of google is to not use the operators on `std::atomic<>`.  Code search `\.load\(`.  There is technically an order argument that is omitted with the operators too.

Reply:
Like this?

Reply:
yep, lookin good